### DISCUSSION TITLE
How do we represent the various users and account types?

### DISCUSSION THREAD

**Discussion by JonruAlveus** [octokit/octokit.net#2504] [Q&A]
I suspect that this Repo was started when there was a bigger difference between and organisation and a personal account.

From what I can make out, for personal accounts there is free and pro tiers, and for organisations there are free, team and enterprise (and enterprise server).

My main question is are we handling these differences properly and how much do we need to ensure coverage for them?

**Comment by timrogers** [comment]
Can you give an example of what you have in mind here? I'm not sure I understand what differences you're thinking about.

**Reply by JonruAlveus** [reply]
Sorry I tend to ramble on a bit sometimes 😆 What I mean is should we be executing tests against specific user types and validating that we are handling the result properly? For example for free users almost everything is only available on public repos. Should we test that creating branch protection on a private repo will fail for a free user but pass for a paid user/organisation. Or should we have a different set of clients and models for free users with less functionality?

**Reply by timrogers** [reply]
It sounds like some of those things might really be testing the behaviour of the GitHub API (which is not fixed and could change - e.g. GitHub could add branch protection for free users) rather than testing the client. What do you think?
